packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1390,6 +1411,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
           spotOptions: props.spotPrice !== undefined ? { maxPrice: parseFloat(props.spotPrice) } : undefined,
           blockDevices: props.blockDevices,
           instanceProfile,
+          ...keyUsageType,
```

Comment:
```suggestion
          keyName: props.keyPair?.keyPairName ?? props?.keyName,
```
Given the above validation, we can simplify this assignment.

Reply:
@lpizzinidev The idea of ​​creating the `keyUsageType` was to send just one of them (keyPair or keyName) to the LaunchTemplate, to avoid sending the `keyName` if `keyPair` is set

```ts
let keyUsageType: { keyPair?: ec2.IKeyPair; keyName?: string } = {};

if (props.keyPair) {
  keyUsageType = { keyPair: props.keyPair };
} else {
  keyUsageType = { keyName: props.keyName };
}
```

It makes sense ?